### DISCUSSION TITLE
3DS Step-up Retry is live!

### DISCUSSION THREAD

**Discussion by sai-harsha-vardhan** [juspay/hyperswitch#2924] [Announcements]
Enhancing your payment experience 🌟

- **Hyperswitch now supports automatic retries of payments failing with 'FRAUD_DETECTED' errors at a payment processor.**
- **This is done by stepping-up the auth type to ‘3DS auth’ in the second automatic attempt through the same connector.**
- **3DS Step-up Retry increases the payments success rates while letting the business relax about chargebacks liability.** 

Your success rate just got even better 🚀✨

**Comment by euluizaraujo** [comment]
Hello how are you? @sai-harsha-vardhan 


Is this configurable via the panel or via the API? Could you provide documentation or more details on how to use this functionality? Is the smart retry functionality 100% available opensource or only for payments do we reject due to fraud_detected?

Is this configurable via the panel or via the API?

**Comment by sai-harsha-vardhan** [comment]
Hi @euluizaraujo,
We're happy to announce that the smart retry functionality is now completely open source. You can find the documentation for it here: https://hyperswitch.io/docs/features/smartRetries

Currently, this feature is configurable via the Config API. Two configurations must be enabled for a merchant to start the auto-retry feature:
1. `key` should_call_gsm_{merchant_id} `value` true
2. `key` max_auto_retries_enabled_{merchant_id} `value` any positive integer

Based on these configurations, if a connector is enabled for retry, we will attempt to retry the payment with the next available connector, following your routing configuration

**Comment by euluizaraujo** [comment]
hi @sai-harsha-vardhan 
 In this documentation sent, it does not talk about the retry scenario in the case of potential fraud, in the documentation I stated that in this case the retry is not performed. Could you explain how it works in this case? (3DS Step-up Retry)

I would like to help evolve this functionality, but we still don't quite understand the technical logic behind it (response, calls), is it possible to put this in the doc or here?

**Comment by sai-harsha-vardhan** [comment]
Hey @euluizaraujo, To provide you with an overview of how it works: it operates similarly to auto-retries. It looks up the connector, error code, message, and flow from GSM. It checks if "is_stepup_possible" and retries are enabled for the merchant. If yes, instead of retrying with another connector, it updates the authentication type to "three_ds" and retries with the same connector.

Detailed documentation on 3DS step-up retries will be available by next week.

**Reply by euluizaraujo** [reply]
@sai-harsha-vardhan  Perfect, thank you very much for your feedback, I await the documentation.

**Reply by sai-harsha-vardhan** [reply]
@euluizaraujo You can access the docs here - https://opensource.hyperswitch.io/features/3ds-step-up-retries
